pkg/apis/samples/v1/types.go

```diff
@@ -424,9 +428,23 @@ func (s *Config) ClusterNeedsCreds() bool {
 		s.Spec.ManagementState == operatorv1.Unmanaged {
 		return false
 	}
-	if s.Status.Conditions == nil {
+	if s.Status.Conditions == nil || len(s.Status.Conditions) == 0 {
 		return true
```

Comment:
why doesn't this return value care what the samples registry is?

Reply:
(seems like you can add an early check that checks what the samples registry even is.  if it's not registry.redhat.io you can short circuit return false)

Reply:
addressed in new push